src/datablock/block.jl

```diff
@@ -155,3 +155,49 @@ function checkblock(
 end
 
 mockblock(block::Keypoints{N}) where N = rand(SVector{N, Float32}, block.sz)
+
+
+# TableRow
+
+"""
+    TableRow{M, N}(catcols, contcols, categorydict) <: Block
+
+`Block` for table rows with M categorical and N continuous columns. `data`
+is valid if it satisfies the `AbstractRow` interface in Tables.jl, values 
+present in indices for categorical and continuous columns are consistent, 
+and `data` is indexable by the elements of `catcols` and `contcols`.
+"""
+struct TableRow{M, N} <: Block
+    catcols::NTuple{M}
+    contcols::NTuple{N}
+    categorydict
+end
```

Comment:
Not sure if FastAI.jl prefers to not use concrete fields @lorenzoh 

```suggestion
struct TableRow{M, N, T} <: Block
    catcols::NTuple{M}
    contcols::NTuple{N}
    categorydict::T
end
```